src/ngraph/pass/manager.cpp

```diff
@@ -122,6 +131,19 @@ void ngraph::pass::Manager::run_passes(shared_ptr<Function> func)
             }
         }
         index++;
+        pass_timer.stop();
+        if (profile_enabled)
+        {
+            PassBase* p = pass.get();
+            string name = typeid(*p).name();
```

Comment:
do you think we could get rid of `m_pass_names` for good and just use this logic to get pass names?

Reply:
yes

Reply:
No, too big a change. Needs a separate PR.